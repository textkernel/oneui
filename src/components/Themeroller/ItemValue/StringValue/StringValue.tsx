import * as React from 'react';
import { Input } from '../../../Input';
import { ThemeStringItem } from '../../../../themes/themerollerConfig';

type Props = {
    item: ThemeStringItem;
    onChange: (value: ThemeStringItem) => void;
};

export const StringValue: React.FC<Props> = ({ item, onChange }) => {
    const handleOnChange = (event) => {
        onChange({
            ...item,
            value: event.target.value,
        });
    };
    return <Input type="text" size="small" value={item.value} onChange={handleOnChange} />;
};

StringValue.displayName = 'StringValue';

StringValue.defaultProps = {};
