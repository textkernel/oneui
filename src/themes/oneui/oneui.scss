@import './base-variables/colors';
@import './base-variables/typography';
@import './base-variables/sizing';
@import './base-variables/misc';
@import './base-variables/header_footer';

@import './mixins/colors';
@import './placeholders/input';
