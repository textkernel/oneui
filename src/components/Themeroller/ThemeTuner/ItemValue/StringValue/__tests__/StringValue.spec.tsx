import React from 'react';
import { render, screen, RenderResult } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { ThemeStringItem } from '@textkernel/oneui';
import { StringValue } from '../StringValue';

describe('StringValue component', () => {
    const onChangeMock = jest.fn();
    const item = { value: 'flex' } as ThemeStringItem;

    let view: RenderResult;

    it.skip('should render component correctly', () => {
        view = render(<StringValue item={item} onChange={onChangeMock} ariaRole="string" />);

        expect(view.container).toMatchSnapshot();
        expect(screen.getByRole('string')).toHaveAttribute('value', 'flex');
    });

    it.skip('should invoke onChange callback when input is changed', async () => {
        const inputValue = { value: 'inline-block' } as ThemeStringItem;
        const user = userEvent.setup();
        render(<StringValue item={item} onChange={onChangeMock(inputValue)} />);

        await user.click(screen.getByRole(''));

        expect(onChangeMock).toHaveBeenCalledWith(inputValue);
    });
});
