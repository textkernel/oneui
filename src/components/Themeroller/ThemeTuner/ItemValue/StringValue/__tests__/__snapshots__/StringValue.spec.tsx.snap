// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StringValue component should render component correctly 1`] = `
<div>
  <input
    class="Input Input--size_small"
    data-lpignore="true"
    type="text"
    value="flex"
  />
</div>
`;
